.NET/Microsoft.Recognizers.Text.DateTime.Tests/English/TestDateExtractor.cs

```diff
@@ -1,97 +1,142 @@
+            BasicTest("I'll go back 1-Jan", 13, 5);
+            BasicTest("I'll go back 28-Nov", 13, 6);
+            BasicTest("I'll go back Wed, 22 of Jan", 13, 14);
+
+            BasicTest("I'll go back Jan first", 13, 9);
+            BasicTest("I'll go back May twenty-first", 13, 16);
+            BasicTest("I'll go back May twenty one", 13, 14);
+            BasicTest("I'll go back second of Aug", 13, 13);
+            BasicTest("I'll go back twenty second of June", 13, 21);
+
+            BasicTest("I'll go back on Friday", 16, 6);
+            BasicTest("I'll go back Friday", 13, 6);
+            BasicTest("I'll go back today", 13, 5);
+            BasicTest("I'll go back tomorrow", 13, 8);
+            BasicTest("I'll go back yesterday", 13, 9);
+            BasicTest("I'll go back the day before yesterday", 13, 24);
+            BasicTest("I'll go back the day after tomorrow", 13, 22);
+            BasicTest("I'll go back the next day", 13, 12);
+            BasicTest("I'll go back next day", 13, 8);
+            BasicTest("I'll go back this Friday", 13, 11);
+            BasicTest("I'll go back next Sunday", 13, 11);
+            BasicTest("I'll go back last Sunday", 13, 11);
+            BasicTest("I'll go back last day", 13, 8);
+            BasicTest("I'll go back the last day", 13, 12);
+            BasicTest("I'll go back the day", 13, 7);
+            BasicTest("I'll go back this week Friday", 13, 16);
+            BasicTest("I'll go back next week Sunday", 13, 16);
+            BasicTest("I'll go back last week Sunday", 13, 16);
+            BasicTest("I'll go back 15 June 2016", 13, 12);
+            BasicTest("a baseball on may the eleventh", 14, 16);
+
+            BasicTest("I'll go back the first friday of july", 13, 24);
+            BasicTest("I'll go back the first friday in this month", 13, 30);
+
+            BasicTest("I'll go back two weeks from now", 13, 18);
+
+            BasicTest("I'll go back next week on Friday", 13, 19);
+            BasicTest("I'll go back on Friday next week", 13, 19);
+
+            BasicTest("past Monday", 0, 11);
+        }
+
+        [TestMethod]
+        public void TestDateExtractAgoLater()
+        {
+            BasicTest("I went back two months ago", 12, 14);
+            BasicTest("I'll go back two days later", 13, 14);
+            BasicTest("who did i email a month ago", 16, 11);
+        }
+
+        [TestMethod]
+        public void TestDateExtractForThe()
+        {
+            BasicTest("I went back for the 27", 12, 10);
+            BasicTest("I went back for the 27th", 12, 12);
+            BasicTest("I went back for the 27.", 12, 10);
+            BasicTest("I went back for the 27!", 12, 10);
+            BasicTest("I went back for the 27 .", 12, 10);
+            BasicTest("I went back for the 21st", 12, 12);
+            BasicTest("I went back for the 22nd", 12, 12);
+            BasicTest("I went back for the second", 12, 14);

```

Comment:
Can you please add a TestMethod for some negative cases?

Reply:
done